Improve and extend launch file tutorial (#1109)
ros2/ros2_documentation: source/Tutorials/Launch-Files/Creating-Launch-Files.rst

.. tabs::

>   .. group-tab:: XML launch file, Eloquent and later

Reviewer: ```suggestion
  .. group-tab:: XML launch file, Eloquent and newer
```

2.1 Examine the launch file
~~~~~~~~~~~~~~~~~~~~~~~~~~~

These import statements pull in some Python ``launch`` modules.
Launch files begin with a launch description:

> .. code-block:: python

Reviewer: I kind of feel like this needs a group-tab.

Actually, I'm really wondering if we want to do this with tabs at all.  It seems to me that this would be significantly easier to follow if it was separate pages for the XML launch files vs. the Python ones.  That is, I would think we would have a top-level tutorial that said something like: "You can write launch files in either XML or Python.  If you want to write an XML one, follow the link to XML, or if you want to write a Python one, follow the link to Python one".  This would get rid of a lot of the group tabs in here and overall make it easier to follow.  This will also make it easier to port to the Rolling branch afterwards.

Author: > Actually, I'm really wondering if we want to do this with tabs at all. It seems to me that this would be significantly easier to follow if it was separate pages for the XML launch files vs. the Python ones. That is, I would think we would have a top-level tutorial that said something like: "You can write launch files in either XML or Python. If you want to write an XML one, follow the link to XML, or if you want to write a Python one, follow the link to Python one"
This would get rid of a lot of the group tabs in here and overall make it easier to follow. This will also make it easier to port to the Rolling branch afterwards.

I integrated both in one tutorial as I thought both versions were "conceptually similar".
It's a bit hard to read because the tabs are also used for distro versioning though.

I'm not sure if we are going to keep using tabs for distro versioning, or if we're going to use one branch per distro or something like that.
In the fist case, I agree that splitting it in XML and Python is going to make things easier.

Reviewer: > I'm not sure if we are going to keep using tabs for distro versioning, or if we're going to use one branch per distro or something like that.

The plan is to move to one branch per ROS 2 distribution.  That will go in effect starting Monday.

> I integrated both in one tutorial as I thought both versions were "conceptually similar".
> In the fist case, I agree that splitting it in XML and Python is going to make things easier.

I agree that they are conceptually similar, but they are very textually different.  Putting both of those things in one page seems to be hiding the information.  If they are separate pages, then it is easier to see them in the tutorial listing.  So I still think it makes sense to split this whole thing up.

Reviewer: Just FYI, we've now switched to one distribution per branch now.  This PR conflicts with the master now.  My suggestion here is to close this PR and rewrite like we talked about above.  I'll leave this open for now, though.

3 ros2 launch
^^^^^^^^^^^^^

> Open in a terminal the directory where you created the launch file, then you can run the following command:

Reviewer: ```suggestion
In a terminal, change to the directory where you created the launch file, then run the following command:
```
